File: LeetCode/Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int> &num) {
        std::map<int, bool> map;
        for (int i = 0; i < num.size(); ++i)
            if (map.find(num[i]) == map.end())
                map[num[i]] = true;

        int k, cnt, ans = INT_MIN;
        for (int i = 0; i < num.size(); ++i)
        {
            if (map.find(num[i]) != map.end() && map.find(num[i] - 1) == map.end()){
                k = i; cnt = 1;
                while (map.find(num[k] + 1) != map.find()){
                    k++;
                    cnt++;
                }
                ans = (ans >= cnt)?ans:cnt;
            }
        }
        return ans;
    }
};

File: LeetCode/Longest Consecutive Sequence.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def longestConsecutive(self, num):
        max_len = 0
        num = set(num)
        for item in num:
            l = 0
            left = item-1
            right = item+1
            while left in num:
                l+=1
                num.remove(left)
                left=left-1

            while right in num:
                l+=1
                num.remove(right)
                right=right+1

            num.remove(item)

            if l > max_len:
                max_len = l
        return max_len


if __name__ == '__main__':
    s=Solution()
    num = [100, 4, 200, 1, 3, 2]
    print s.longestConsecutive(num)